lib/benchee/conversion/scale.ex

```diff
@@ -81,6 +81,8 @@ defmodule Benchee.Conversion.Scale do
       iex> Benchee.Conversion.Scale.scale 12345, unit
       12.345
   """
+  def scale(nil, _), do: 0.0
```

Comment:
sure we want this? Seems like it'd possibly hide errors where nil values end up  here. I don't think `nil` values should ever end up here or what am I missing? :)

Reply:
I honestly can't remember the context around when I first put this in, but when I get rid of this, the tests that fail are around protecting against weird things that probably shouldn't happen but do:

```

  1) test .format_scenarios it doesn't blow up if some come back with a median et. al. of nil (Benchee.Formatters.Console.ReductionsTest)
     test/benchee/formatters/console/reductions_test.exs:465
     ** (ArithmeticError) bad argument in arithmetic expression
     code: Reductions.format_scenarios(scenarios, %{
     stacktrace:
       (benchee 1.0.1) lib/benchee/conversion/scale.ex:86: Benchee.Conversion.Scale.scale/2
       (benchee 1.0.1) lib/benchee/formatters/console/helpers.ex:46: Benchee.Formatters.Console.Helpers.count_output/2
       (benchee 1.0.1) lib/benchee/formatters/console/reductions.ex:215: Benchee.Formatters.Console.Reductions.reference_report/3
       (benchee 1.0.1) lib/benchee/formatters/console/reductions.ex:202: Benchee.Formatters.Console.Reductions.comparison_report/5
       (benchee 1.0.1) lib/benchee/formatters/console/reductions.ex:53: Benchee.Formatters.Console.Reductions.render/2
       test/benchee/formatters/console/reductions_test.exs:490: (test)

.

  2) test .format_scenarios it doesn't blow up if some of the values have no statistics (yes that happened) (Benchee.Formatters.Console.ReductionsTest)
     test/benchee/formatters/console/reductions_test.exs:332
     ** (ArithmeticError) bad argument in arithmetic expression
     code: Reductions.format_scenarios(scenarios, %{
     stacktrace:
       (benchee 1.0.1) lib/benchee/conversion/scale.ex:86: Benchee.Conversion.Scale.scale/2
       (benchee 1.0.1) lib/benchee/formatters/console/helpers.ex:46: Benchee.Formatters.Console.Helpers.count_output/2
       (benchee 1.0.1) lib/benchee/formatters/console/reductions.ex:226: anonymous fn/3 in Benchee.Formatters.Console.Reductions.comparisons/3
       (elixir 1.10.0) lib/enum.ex:1396: Enum."-map/2-lists^map/1-0-"/2
       (benchee 1.0.1) lib/benchee/formatters/console/reductions.ex:203: Benchee.Formatters.Console.Reductions.comparison_report/5
       (benchee 1.0.1) lib/benchee/formatters/console/reductions.ex:53: Benchee.Formatters.Console.Reductions.render/2
       test/benchee/formatters/console/reductions_test.exs:357: (test)
```

So leads me towards wanting to keep this, at least for now.

Reply:
Iirc the tests are just copied from the memory formatter and that one deals with it in its own way:

```elixir
  defp memory_output(nil, _unit), do: "N/A"

  defp memory_output(memory, unit) do
    Memory.format({Memory.scale(memory, unit), unit})
  end
```

I'd prefer the same for reductions tbh. I don't think that handling these errors should be in conversion/the unit part (partly because I still want to extract it and keep benchee specific handling out).

At the same time, it leaves room for warning the user or deciding on an output closer to the source.

Also, speaking of other things I want to do - I'd like to to deduplicate the formatting and it'd help for it to be more uniform :)

Reply:
Yeah, cool - I moved that error handling back up into the console formatter for this stuff. The cases are pretty much impossible, but it's good to still handle these.